gensim/models/sent2vec.py

```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+from __future__ import division
+import logging
+from collections import defaultdict
+from six import itervalues, string_types
+import numpy as np
+from numpy import dot
+from gensim import utils, matutils
+
+from gensim.models.word2vec import Word2Vec
+from gensim.models.fasttext import FastText
+from gensim.models.wrappers.fasttext import FastTextKeyedVectors
+from gensim.models.wrappers.fasttext import FastText as Ft_Wrapper
+from random import randint
+
+logger = logging.getLogger(__name__)
+
+MAX_WORDS_IN_BATCH = 10000
+
+class Sent2Vec(FastText):
+    def __init__(
+            self, sentences=None, sg=0, hs=0, size=100, alpha=0.2, window=5, min_count=5,
+            max_vocab_size=None, word_ngrams=2, loss='ns', sample=1e-3, seed=1, workers=3, min_alpha=0.0001,
+            negative=5, cbow_mean=1, hashfxn=hash, iter=5, null_word=0, min_n=3, max_n=6, sorted_vocab=1, bucket=2000000,
+            trim_rule=None, batch_words=MAX_WORDS_IN_BATCH, dropoutK=2):
+
+        # sent2vec specific params
+        #dropoutK is the number of ngrams dropped while training a sent2vec model
+        self.dropoutK = dropoutK
+
+        super(Sent2Vec, self).__init__(bucket=bucket, word_ngrams=word_ngrams, min_n=min_n, max_n=max_n,
+            sentences=sentences, size=size, alpha=alpha, window=window, min_count=min_count,
+            max_vocab_size=max_vocab_size, sample=sample, seed=seed, workers=workers, min_alpha=min_alpha,
+            sg=sg, hs=hs, negative=negative, cbow_mean=cbow_mean, hashfxn=hashfxn, iter=iter, null_word=null_word,
+            trim_rule=trim_rule, sorted_vocab=sorted_vocab, batch_words=batch_words)
+
+    def scan_vocab(self, sentences, progress_per=10000, trim_rule=None):
+        """Do an initial scan of all words appearing in sentences."""
```

Comment:
Please add more description in docstring, what's a difference between fasttext and sent2vec.